src/validator/Tests.java

```diff
@@ -0,0 +1,55 @@
+package validator;
+
+import static org.junit.Assert.*;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+import org.junit.Test;
+
+public class Tests {
```

Comment:
This is basically one huge integration test for a code that is not really testable. Let's try splitting the code according to my comments from this PR, creating nice, small unit tests for the logic and then refactoring this code, probably using a @SpringBootTest